src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -96,4 +89,36 @@ else if (!assessor.equals(reviewer) && (originalComplaint.getComplaintType() !=
         complaintResponse.setComplaint(originalComplaint);
         return complaintResponseRepository.save(complaintResponse);
     }
+
+    /**
+     * Checks whether the reviewer is authorized to respond to this complaint
+     *
+     * 1. Team Exercises
+     *    => The team tutor assesses the submissions and responds to complaints and more feedback requests
+     *
+     * 2. Individual Exercises
+     *    => Complaints can only be handled by a tutor who is not the original assessor
+     *    => More feedback requests are handled by the assessor himself
+     *
+     * @param participation Participation to which the complaint belongs to
+     * @param complaint Complaint for which to check
+     * @param assessor Assessor of the submission whose assessment was complained about
+     * @param reviewer Reviewer who is trying to create a response to the complaint
+     * @return true if the tutor is allowed to respond to the complaint, false otherwise
+     */
+    private boolean isUserAuthorizedToRespondToComplaint(StudentParticipation participation, Complaint complaint, User assessor, User reviewer) {
```

Comment:
I'm not sure if we want to allow instructors to review every complaint, is that something you considered?

Reply:
Good point. I basically wanted to keep the existing behavior for individual exercises the same and model the behavior for team exercises after it as far as there are parallels.

The behavior for individual exercises is that instructors could respond to a complaint but not to a more feedback request. That makes sense in so far as complaints in individual exercises can be handled by any secondary person who is at least a tutor while more feedback requests can only be handled by the originally assessor (since only they know why they assessed something a certain way).

In team exercises, both types of student feedback are only handled by the team tutor himself so in that regard they both behave more like "more feedback" requests for individual exercises. Thus I think it makes sense if only the original assessor can respond to complaints for team exercises. It would be surprising for the students if suddenly a different person responds when they expect it to be their tutor for team exercises.

Reply:
My idea was that instructors could possibly solve a complaint in case a tutor is unable to do so, either because of lack of knowledge or e.g. personal issues that prevent the tutor from evaluating the complaint/feedback request.

I agree that it does not make sense to add this right now, as there is also no way to display these requests. I still believe there might be some scenarios, like mentioned above, where it might be valid for an instructor to solve a complaint.

As this is out-of-scope for this PR, I will resolve this discussion.